test/bucket.js

```diff
@@ -60,22 +60,11 @@ function FakeNotification(bucket, id) {
   this.id = id;
 }
 
-const requestCached = request;
-let requestOverride;
-function fakeRequest() {
-  return (requestOverride || requestCached).apply(null, arguments);
+const fetchCached = fetch;
```

Comment:
If we're going to rewrite this code, I am personally *much* more fond of using `nock` if mocking the http transport library.  That way we're protected from whatever future changes we choose to make here.  Really interested to hear others thoughts :)

Reply:
Makes sense. Let's introduce a different mocking library in a separate PR.